code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -606,6 +611,95 @@ GLOBAL_VAR(medibot_unique_id_gen)
 /obj/machinery/bot_core/medbot
 	req_one_access = list(ACCESS_MEDICAL, ACCESS_ROBOTICS)
 
+
+/mob/living/simple_animal/bot/medbot/cola
+	name = "Sponsored Medibot"
+	desc = "Even medical isn't safe from corporate greed. May contain cola."
+	icon_state = "colabot0"
+	radio_key = /obj/item/encryptionkey/medical_sponsor  // grants cargo
+	COOLDOWN_STATIC_DECLARE(fund)
+	var/static/budget = 1000  // initial value (will get refilled instantly)
+	var/static/refill = 1000
+	var/payout = 250  // at start, uses entire budget in 8 heals
+
+/mob/living/simple_animal/bot/medbot/cola/New()
+	START_PROCESSING(SSobj, src)
+
+/mob/living/simple_animal/bot/medbot/cola/process()
+	if(COOLDOWN_FINISHED(src, fund))
+		refill()
+		COOLDOWN_START(src, fund, 2 MINUTES)
+
+/obj/item/mop/advanced/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/mob/living/simple_animal/bot/medbot/explode()
+	. = ..()
+	new /obj/item/reagent_containers/food/drinks/soda_cans/cola(drop_location())
+
+/mob/living/simple_animal/bot/medbot/cola/update_icon()
```

Comment:
use update_appearance if possible

Reply:
this is copied from the base medbot it should be alr